axelor-human-resource/src/main/java/com/axelor/apps/hr/service/allocation/AllocationLineComputeServiceImpl.java

```diff
@@ -140,49 +141,56 @@ public BigDecimal computePlannedTime(
     BigDecimal totalPlannedTime = BigDecimal.ZERO;
     if (fromDate == null
         || toDate == null
-        || project == null
-        || !project.getIsShowPlanning()
         || !Optional.ofNullable(appProjectService.getAppProject())
             .map(AppProject::getEnablePlanification)
             .orElse(false)) {
       return totalPlannedTime;
     }
-
     List<ProjectPlanningTime> projectPlanningTimeList = new ArrayList<>();
-    if (employee == null) {
-      projectPlanningTimeList =
-          planningTimeRepo.findByProjectAndPeriod(project, fromDate, toDate).fetch();
-    } else {
-      projectPlanningTimeList =
-          planningTimeRepo
-              .findByEmployeeProjectAndPeriod(employee, project, fromDate, toDate)
-              .fetch();
+    projectPlanningTimeList = getProjectPlanningTimeList(fromDate, toDate, employee, project);
+
+    if (ObjectUtils.isEmpty(projectPlanningTimeList)) {
+      return totalPlannedTime;
     }
+    Unit dayUnit = appBaseService.getAppBase().getUnitDays();
+
+    for (ProjectPlanningTime projectPlanningTime : projectPlanningTimeList) {
+      BigDecimal plannedTime =
+          unitConversionForProjectService.convert(
```

Comment:
Before every unitConversionForProjectService.convert, check that the project.NumberHoursADay is > 0, else return null as project